src/lists.c

```diff
@@ -2610,6 +2610,7 @@ static Int InitKernel (
         }
     }
 
+    SetupFlagsListsAsLists();
```

Comment:
Adding the API turned out to be a bit more trouble than I expected. I was trying to avoid storing TRUES_FLAG as a PLIST (waste of memory) or computing it on the fly each time (waste of CPU), so I thought "why not simply let the flags list (which is simply a packed version of its TRUES_FLAGS now) just be the list".  It's not a lot of code in the end, but it's a lot  of trivial functions and table entries.

Reply:
So, what is gained by all these complications? I mean, for most uses, the overhead of `TRUES_FLAG` (be it computational or memory wise) is irrelevant (e.g. when viewing flags). The only really crucial places using it is `RunImmediates`, which ironically is the one place in this PR where the overhead is not really avoided ;-) . Well, perhaps `RankFilter` also is performance critically, but it could easily be reimplemented in C, too.

So unless there is some clear win in having this list interface for flags, I'd rather remove it again, and keep this PR small(er)...

And I am really curious if implementing and using `APPEND_TRUES_FLAG` helps at all... and whether perhaps we should reimplement `RunImmediates` in C anyway...

BTW, did you perform any kind of profiling before/after to see where the bottlenecks are resp. what shifted?

Reply:
I haven't done any profiling yet. Ran out of time. If you're set up to get cachegrind output for startup, that would save me the trouble of creating it.

Reworking RunImmediates and WITH_IMPS_FLAGS to be sensible implementations given this data structure was next on the agenda.

APPEND_TRUES_FLAG does seem like a potentially better answer, though, in which case much of this PR can go.  

It would be quite nice to get TRUES_FLAG off the critical path so that we don't have to cache the result, though.